src/System.Linq.Expressions/src/System/Linq/Expressions/BlockExpression.cs

```diff
@@ -899,8 +909,6 @@ public static BlockExpression Block(Type type, IEnumerable<ParameterExpression>
                 {
                     var lastExpression = expressionList[expressionCount - 1];
 
-                    ContractUtils.RequiresNotNull(lastExpression, nameof(expressions));
```

Comment:
Moved `RequiresCanRead` higher up, which incorporates this check. It has the effect of validating expressions left-to-right, so we get an exception referring to the first expression slot rather than potentially an error for the last expression first.